src/oauth2.js

```diff
@@ -23,7 +23,11 @@ const listenForCredentials = (popup, state, resolve, reject) => {
     }
 
     if (response.access_token) {
-      resolve(response.access_token)
+      const result = {
+        token: response.access_token,
+        expires_at: response.expires_in ? new Date().getTime() + parseInt(response.expires_in) * 1000 : null
```

Comment:
This would violate the OAuth2 spec. But I added a `isNaN` check just to make sure.